replica/database.cc

```diff
@@ -1647,31 +1647,34 @@ future<> dirty_memory_manager::flush_when_needed() {
                 // Do not wait. The semaphore will protect us against a concurrent flush. But we
                 // want to start a new one as soon as the permits are destroyed and the semaphore is
                 // made ready again, not when we are done with the current one.
-                (void)this->flush_one(mtlist, std::move(permit)).then([r]() {
+                (void)this->flush_one(mtlist, std::move(permit)).then_wrapped([this, r](future<> f) {
```

Comment:
I don't understand why you wanted to change this code. The old code had then() and handle_exception() - the new code has one then_wrapped() for both. Did you do this because it's more efficient? Or what? It's not explained in the commit message.